Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -8728,6 +8729,33 @@ bool WebPage::shouldSkipDecidePolicyForResponse(const WebCore::ResourceResponse&
     return true;
 }
 
+void WebPage::registerRedirectionToFollow(WebCore::ResourceResponse&& response)
+{
+    auto* localMainFrame = dynamicDowncast<LocalFrame>(m_page->mainFrame());
+    if (!localMainFrame) {
+        WEBPAGE_RELEASE_LOG_ERROR(Loading, "WebPage::registerRedirectionToFollow failed without frame");
+        return;
+    }
+
+    auto* loader = localMainFrame ? localMainFrame->loader().policyDocumentLoader() : nullptr;
```

Comment:
/* Unofficial Review */
localMainFrame should not be a nullptr here and below. It is checked above.
auto* loader = localMainFrame->loader().policyDocumentLoader() is enough.

Reply:
done